src/Sylius/Behat/Context/Api/Admin/ManagingTaxRateContext.php

```diff
@@ -0,0 +1,524 @@
+    {
+        if ($zone) {
+            $this->client->addRequestData('zone', $this->iriConverter->getIriFromItem($zone));
+        }
+    }
+
+    /**
+     * @When I make it applicable for the :taxCategory tax category
+     * @When I change it to be applicable for the :taxCategory tax category
+     * @When I do not specify related tax category
+     */
+    public function iMakeItApplicableForTheTaxCategory(?TaxCategoryInterface $taxCategory = null): void
+    {
+        if ($taxCategory) {
+            $this->client->addRequestData('category', $this->iriConverter->getIriFromItem($taxCategory));
+        }
+    }
+
+    /**
+     * @When I specify its amount as :amount%
+     * @When I do not specify its amount
+     */
+    public function iSpecifyItsAmountAs(?string $amount = null): void
+    {
+        if ($amount) {
+            $this->client->addRequestData('amount', $amount);
+        }
+    }
+
+    /**
+     * @When I choose the default tax calculator
+     */
+    public function iChooseTheDefaultTaxCalculator(): void
+    {
+        $this->client->addRequestData('calculator', 'default');
+    }
+
+    /**
+     * @When I make it start at :startDate and end at :endDate
+     */
+    public function iMakeItStartAtAndEndAt(string $startDate, string $endDate): void
+    {
+        $this->client->addRequestData('startDate', $startDate);
+        $this->client->addRequestData('endDate', $endDate);
+    }
+
+    /**
+     * @When I set the start date to :startDate
+     */
+    public function iSetTheStartDateTo(string $startDate): void
+    {
+        $this->client->addRequestData('startDate', $startDate);
+    }
+
+    /**
+     * @When I set the end date to :endDate
+     */
+    public function iSetTheEndDateTo(string $endDate): void
+    {
+        $this->client->addRequestData('endDate', $endDate);
+    }
+
+    /**
+     * @When I add it
+     * @When I try to add it
+     */
+    public function iAddIt(): void
+    {
+        $this->client->create();
+    }
+
+    /**
+     * @Then I should be notified that it has been successfully created
+     */
+    public function iShouldBeNotifiedThatItHasBeenSuccessfullyCreated(): void
+    {
+        Assert::true(
+            $this->responseChecker->isCreationSuccessful($this->client->getLastResponse()),
+            'Tax tax rate could not be created'
+        );
+    }
+
+    /**
+     * @Then the tax rate :taxRate should appear in the registry
+     * @Then I should see the tax rate :taxRate in the list
+     */
+    public function theTaxRateShouldAppearInTheRegistry(TaxRateInterface $taxRate): void
+    {
+        $name = $taxRate->getName();
+
+        Assert::true(
+            $this->responseChecker->hasItemWithValue($this->client->index(Resources::TAX_RATES), 'name', $name),
+            sprintf('Tax rate with name %s does not exists', $name)
+        );
+    }
+
+    /**
+     * @Given I choose "Included in price" option
```

Comment:
Do we still order steps by its type? /cc @Zales0123 @GSadee

Reply:
Yeah, this step should be extracted to the setup class as anywhere else. I missed that.